docs/koji.md

```diff
@@ -0,0 +1,313 @@
+mount | grep /mnt/koji && mount | grep /mnt/tmp && systemctl start kojid
+mount | grep /mnt/koji && mount | grep /mnt/tmp && systemctl start kojira
+</pre>
+
+On our current VM flavour there is a local SSD NVMe storage (/dev/nvme0n1) with two partitions created (50/50). The first one is swap and the second one is mounted as /mnt/tmp where koji does all the work. This volume needs to be fast, it grows over the time and contains temporary files (built packages, build logs, support files).
+
+The main data folder where PostgreSQL database and koji generated repositories and external repositories are present is on EBS volume mounted as /mnt/koji. Note this was created as ext4 which can sometimes lead to mkfs, perhaps xfs would be better fit for our use case.
+
+Services required for koji (postgresql, httpd, kojid, kojira, rsyncd) are only started if required volumes are mounted.
+
+### Ports and Security Group
+
+The instance must be running in a security group with ports:
+
+ * 22
+ * 80
+ * 443
+ * 873 (rsyncd)
+ * 44323 (read only monitoring microsite) allowed (all IPv4 TCP).
+
+
+### Cache and logs
+
+Demanding directories from root volume which is small to /mnt/tmp scratch disk and created symlinks for those:
+
+* /var/cache/mrepo -> /mnt/tmp/cache/mrepo
+* /var/log/httpd -> /mnt/tmp/log/httpd
+
+### Hostname
+
+The instance has a floating IP, in /etc/hosts we have an entry for that:
+
+34.224.159.44 koji.katello.org kojihub.katello.org koji kojihub
+
+When the IP changes, make sure this does change as well.
+
+When new instance is booted via AWS, it will have a random hostname assigned. In the rc.local we set the hostname to koji.katello.org on every boot.
+
+### Backups
+
+There is a cron job (/etc/cron.weekly/koji-backup) that performs two actions every week:
+
+Full PostgreSQL database dump into /mnt/koji/backups/postgres.
+
+File system backup of /mnt/tmp (ephemeral storage) into /mnt/koji/backups/ephemeral. This backup skips all files named RPM (these are not needed), duplicity tool is used, no encryption is done. The main purpose of this backup is to store required filesystem structure so koji can be quickly brought up after crash. Since the backup mostly contains directories and build logs, it is not big. To restore that, use:
+
+duplicity restore file:///mnt/koji/backups/ephemeral /mnt/tmp --force --no-encryption
+
+Both backups does not have any rotation and need to be deleted every year. The full backup script looks like:
+
+<pre>
+#!/bin/bash
+/usr/bin/duplicity --full-if-older-than 1M --no-encryption -vWARNING --exclude '/mnt/tmp/**/*rpm' /mnt/tmp file:///mnt/koji/backups/ephemeral
+date=`date +"%Y%m%d"`
+filename="/mnt/koji/backups/postgres/koji_${date}.dump"
+pg_dump -Fc -f "$filename" -U koji koji
+</pre>
+
+### Upgrades
+
+We are running CentOS 7 with Koji (1.11) installed from EPEL7 and mrepo package installed from Fedora. Since koji was bumped in the EPEL to non-compatible version, we have disabled EPEL7 repository for now.
```

Comment:
So this is where I am confused. From `/etc/yum.repos.d/epel.repo`
```init
[epel]
name=Extra Packages for Enterprise Linux 7 - $basearch
#baseurl=http://download.fedoraproject.org/pub/epel/7/$basearch
metalink=https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch
failovermethod=priority
# DISABLED to delay upgrade to new koji-1.13 or newer
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
```

So it's enabled, but has an outdated comment. There's also a `/etc/yum.repos.d/epel.repo.rpmnew` which has an updated metalink.

I've now replaced the old repo file.

```suggestion
We are running CentOS 7 with Koji (1.20) installed from EPEL7 and mrepo installed as an unmanaged standalone file.
```